core/controllers/topic_editor.py

```diff
@@ -412,12 +383,9 @@ def put(self, topic_id):
             raise self.InvalidInputException(
                 'Publish status should only be true or false.')
 
-        try:
```

Comment:
Ditto, this is actually a pretty important condition as both these functions can raise errors under more than one condition.

Reply:
We are testing the value of publish_status and the user rights above. Hence again this statement is unreachable. I have added the case with invalid topic id in the decorator though

Reply:
There is a condition in the function which raises an error when topic is already published which is not checked in the decorator.

Reply:
I meant this exception is raised when the user is not authorised to publish an exploration and this check is done in the decorator

Reply:
Not just that, the exception is also raised if topic_rights doesn't exist.

Reply:
This isn't addressed.

Reply:
Done